src/UI/Art/SpriteArtist/SpriteMapArtist.java

```diff
@@ -227,7 +239,9 @@ private void buildMapImage()
     @Override
     public void receiveMapChangeEvent(MapChangeEvent event)
     {
-      myArtist.buildMapImage();
+      XYCoord xyc = event.getStartPoint();
+      if( null != xyc ) myArtist.redrawBaseTile(xyc); // Redraw one tile if that's all that changed.
+      else myArtist.buildMapImage(); // Redraw the whole map if we have to.
```

Comment:
I feel like this is probably a weird edge case. Why would you only change the weather on one tile?

Reply:
uh... no comment

Reply:
Actually, that's a good point. How about giving the MapChangeEvent a set of coordinates that are changing?
....for that matter, did you check that file in?

Reply:
It might be a decent optimization to use this for buildings as well, and just abolish the TerrainObject idea.
That way, the only sprites that are drawn every frame are the map and units (and UI and stuff obv)